base/Enums.jl

```diff
@@ -140,4 +141,152 @@ macro enum(T,syms...)
     return blk
 end
 
+abstract EnumSet <: AbstractEnum
+
+Base.convert{T<:Integer}(::Type{T}, x::EnumSet) = convert(T, convert(Unsigned, x))
+
+for op in (:|, :&, :xor)
+    @eval function Base.$op{T<:EnumSet}(x::T,y::T)
+        reinterpret(T, ($op)(convert(Unsigned, x), convert(Unsigned, y)))
+    end
+end
+
+Base.union{T<:EnumSet}(x::T, y::T) = x | y
+Base.intersect{T<:EnumSet}(x::T, y::T) = x & y
+Base.issubset{T<:EnumSet}(x::T, y::T) = x & y == x
```

Comment:
On second thought, `issubset` has a nice infix operator too, and does perhaps match closer intuition.